armeria-backend/src/main/scala/sttp/client3/armeria/ArmeriaBackend.scala

```diff
@@ -51,22 +47,28 @@ import scala.io.Source
 import scala.jdk.FutureConverters._
 import scala.util.Try
 
-class ArmeriaBackend(implicit ec: ExecutionContext = ExecutionContext.global) extends SttpBackend[Future, Any] {
+class ArmeriaBackend(client: Option[WebClient] = None)(implicit ec: ExecutionContext = ExecutionContext.global)
+    extends SttpBackend[Future, Any] {
   type PE = Any with capabilities.Effect[Future]
 
   override def send[T, R >: PE](request: Request[T, R]): Future[Response[T]] =
     adjustExceptions(request)(execute(request))
 
   private def execute[T, R >: PE](request: Request[T, R]): Future[Response[T]] =
-    WebClient
-      .builder()
-      .responseTimeout(Duration.ofMillis(request.options.readTimeout.toMillis))
-      .build()
+    getClient(request)
       .execute(requestToArmeria(request))
       .aggregate()
       .asScala
       .flatMap(responseFromArmeria(request, _))
 
+  private def getClient(request: Request[_, _]): WebClient =
+    client.getOrElse(
+      WebClient
+        .builder()
+        .writeTimeout(Duration.ofMillis(request.options.readTimeout.toMillis))
```

Comment:
So to override the read timeout we need to build a new client? This won't allocate any new thread pools / connections - how are connections shared?

It wouldn't be good to allocate a new connection pool for each request

Reply:
Clients are build using default factory which shares a common connection pool with them so it's ok.
However there's a "hack" which allows to reuse instance of WebClient using Armeria delegates mechanism, but the drawback is that the timeout value would have to be stored in headers. Otherwise delegates cannot access it.